velox/exec/ContainerRowSerde.cpp

```diff
@@ -373,37 +372,56 @@ int compareStringAsc(
 }
 
 template <>
-int compare<TypeKind::VARCHAR>(
+std::optional<int32_t> compare<TypeKind::VARCHAR>(
     ByteStream& left,
     const BaseVector& right,
     vector_size_t index,
     CompareFlags flags) {
-  VELOX_DCHECK(!flags.mayStopAtNull(), "not supported null handling mode");
-
   auto result = compareStringAsc(left, right, index, flags.equalsOnly);
   return flags.ascending ? result : result * -1;
 }
 
 template <>
-int compare<TypeKind::VARBINARY>(
+std::optional<int32_t> compare<TypeKind::VARBINARY>(
     ByteStream& left,
     const BaseVector& right,
     vector_size_t index,
     CompareFlags flags) {
-  VELOX_DCHECK(!flags.mayStopAtNull(), "not supported null handling mode");
-
   auto result = compareStringAsc(left, right, index, flags.equalsOnly);
   return flags.ascending ? result : result * -1;
 }
 
+std::optional<std::optional<int32_t>>
```

Comment:
Double optional is a bit hard to reason about. Is this necessary?

Reply:
Reuse `BaseVector::compareNulls` now.